cluster/charts/universal-crossplane/templates/upbound-agent/configmap-deployment-spec.yaml

```diff
@@ -0,0 +1,93 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: upbound-agent-deployment-spec
+  labels:
+    {{- include "labelsAgent" . | nindent 4 }}
+data:
+  spec.yaml: |-
```

Comment:
Ok, actually this solution is not as great as it initially sounded (at least with current implementation) 😅

It allows us to leverage helm templating and an easy way to change agent deployment spec but it is also causing a backdoor violating RBAC. Anyone with write access to configmap can run arbitrary pods by modifying spec here. 
I remember @hasheddan has passed a similar route while he was working on making provider controller deployments configurable. 

Requirements are not same here though, we always want to run agent (not any provider) but still want to keep it configurable (e.g. setting resource limits/request). I am also not willing to introduce a new API type (like a regular k8s operator) since there will always be 0 or 1 instance of agent running and we would need to deal with deploying CRD and CR with helm chart in that case which sounds like an overhead here. May be, we can just get a required/safe subset of deployment spec (similar to ControllerConfigSpec) as input from this configmap rather than expecting whole deployment spec.

Reply:
After some discussion with @muvaf, we agreed on another solution avoiding the problem mentioned above but still lightweight and lets us leverage helm templating.

We will pass templated agent deployment spec as a flag to bootstrapper as base64 encoded.

Reply:
@turkenh @muvaf this is a clever idea! I like it :)